gdscript/sidebar.gd

```diff
@@ -1032,95 +1044,163 @@ enum HistoryListPopupItem {
 			if checked_out_branch.is_main:
 				return
 			# get the root hash from the checked_out_branch
 			previous_heads = checked_out_branch.get("forked_at", PackedStringArray([]))
 		else:
-			previous_heads = [history_list.get_item_metadata(prev_idx)]
+			previous_heads = [prev_item.get_metadata(0)]
 
 		if previous_heads.size() > 0:
 			# diff_section_header.text = DIFF_SECTION_HEADER_TEXT_FORMAT % [prev_change_hash.substr(0, 7), change_hash.substr(0, 7)]
-			var text = history_list.get_item_text(index)
-			var name = text.split(" ")[0].strip_edges()
-			if name == "↪️":
-				name = text.split(" ")[1].strip_edges() + "'s merged branch"
-			var date = text.split("-")[1].strip_edges()
-			diff_section_header.text = "Showing changes from %s - %s" % [name, date]
-			var diff = update_properties_diff(checked_out_branch, ["foo", "bar"], previous_heads, change_heads)
+			var text = selected_item.get_text(1 if DEV_MODE else 0)
+			var date = selected_item.get_text(2 if DEV_MODE else 1)
+			var commit_name = text.split(" ")[0].strip_edges()
+			if commit_name == "↪️":
+				commit_name = text.split(" ")[1].strip_edges() + "'s merged branch"
+			diff_section_header.text = "Showing changes from %s - %s" % [commit_name, date]
+			update_properties_diff(checked_out_branch, 2, previous_heads, change_heads)
 			%ClearDiffButton.visible = true
 			inspector.visible = true
 		else:
 			printerr("no prev change hash")
 	else:
 		printerr("no change hash")
 
-func _on_clear_diff_button_pressed():
-	_on_empty_clicked(Vector2.ZERO, 0)
+# display the default diff, for when there's no available selected diff or if we're merging/reverting
+func update_diff_default(checked_out_branch, history):
+	%ClearDiffButton.visible = false
 
+	var heads_before
+	var heads_after
+	
+	inspector.visible = true
 
-func _on_empty_clicked(_vec2, _idx):
-	%ClearDiffButton.visible = false
-	update_ui(true)
+	if checked_out_branch.is_merge_preview:
+		var source_branch = GodotProject.get_branch_by_id(checked_out_branch.forked_from)
+		var target_branch = GodotProject.get_branch_by_id(checked_out_branch.merge_into)
+		heads_before = checked_out_branch.merge_at
+		heads_after = checked_out_branch.heads
+		diff_section_header.text = "Showing changes for \"" + source_branch.name + "\" -> \"" + target_branch.name + "\""
 
-func update_properties_diff(checked_out_branch, changes, heads_before, heads_after) -> Dictionary:
+	elif checked_out_branch.is_revert_preview:
+		var source_branch = GodotProject.get_branch_by_id(checked_out_branch.forked_from)
+		heads_before = checked_out_branch.forked_at
+		heads_after = checked_out_branch.heads
+		var heads_short_form = heads_to_short_form(checked_out_branch.reverted_to)
+		diff_section_header.text = "Showing changes from \"" + source_branch.name + "\" reverted to \"" + heads_short_form + "\""
 
+	# main branch cannot have content in the diff by default
+	elif checked_out_branch.is_main:
+		inspector.visible = false
+		diff_section_header.text = "Changes"
```

Comment:
we don't show the default diff when we're checked out on main, so we shouldn't run `update_properties_diff`; just return after this
```suggestion
		diff_section_header.text = "Changes"
        return
```